### PR TITLE
HTML-772: Fixed the location id display instead of name when used wit…

### PR DESCRIPTION
**Pull request by mhawila**
…h autocomplete

HTML-772: encounterLocation Tag Displays Location ID When Used with Autocomplete in VIEW Mode

## Issue I worked on

see https://issues.openmrs.org/browse/HTML-772

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api-tests/src/test/java/org/openmrs/module/htmlformentry/widget/AutocompleteWidgetTest.java [hunk 1]
```diff
@@ -80,4 +81,19 @@ public void autocompleteWidget_shouldAcceptLocationOptionsWithSpecialCharacters(
 		}
 		
 	}
+	
+	@Test
+	public void autocompleteWidget_shouldGenerateTheCorrectStringValueForLocationInVIEWMode() throws Exception {
+		AutocompleteWidget autocompleteWidget = new AutocompleteWidget(Location.class);
+		FormEntryContext context = new FormEntryContext(FormEntryContext.Mode.VIEW);
+		
+		Integer initialLocationId = 100;
```
Comment (dkayiwa) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/widget/AutocompleteWidgetTest.java:90]: Is there any reason why you are not declaring these as constants?

Comment (mhawila) [reply api-tests/src/test/java/org/openmrs/module/htmlformentry/widget/AutocompleteWidgetTest.java:90]: Done

Comment (HerbertYiga) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/widget/AutocompleteWidgetTest.java:90]: Add this before your methods with a final key word

File: api-tests/src/test/java/org/openmrs/module/htmlformentry/widget/AutocompleteWidgetTest.java [hunk 2]
```diff
@@ -80,4 +81,19 @@ public void autocompleteWidget_shouldAcceptLocationOptionsWithSpecialCharacters(
...
+		String locationName = "Willa Home Clinic";
```
Comment (HerbertYiga) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/widget/AutocompleteWidgetTest.java:91]: Add this before your methods with a final key word  too

**Pull request merged**